protocol/x/vault/keeper/vault.go

```diff
@@ -0,0 +1,83 @@
+package keeper
+
+import (
+	"math/big"
+
+	"cosmossdk.io/store/prefix"
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/lib/log"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+)
+
+// GetVaultEquity returns the equity of a vault (in quote quantums).
+func (k Keeper) GetVaultEquity(
+	ctx sdk.Context,
+	vaultId types.VaultId,
+) (*big.Int, error) {
+	netCollateral, _, _, err := k.subaccountsKeeper.GetNetCollateralAndMarginRequirements(
+		ctx,
+		satypes.Update{
+			SubaccountId: *vaultId.ToSubaccountId(),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return netCollateral, nil
+}
+
+// DecommissionVaults decommissions all vaults with positive shares and non-positive equity.
+func (k Keeper) DecommissionVaults(
```

Comment:
nit: can we clarify this function name? It sounds like `DecomissionVaults` would decomission all of the vaults, maybe `DecommissionVaultsWithNegativeEquity